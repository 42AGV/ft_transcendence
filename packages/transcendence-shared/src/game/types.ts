export type PlayerIds = {
  playerOneId: string;
  playerTwoId: string;
};
