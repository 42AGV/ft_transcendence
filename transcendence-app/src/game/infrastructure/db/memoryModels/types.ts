export type UserId = string;
export type GameId = string;
