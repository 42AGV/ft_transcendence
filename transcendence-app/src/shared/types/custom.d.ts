import { User } from '../../user/user.domain';

declare module 'http' {
  interface IncomingMessage {
    user: Promise<User | null>;
  }
}
