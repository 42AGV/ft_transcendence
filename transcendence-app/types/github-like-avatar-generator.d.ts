type GenerateAvatarParams = {
  blocks: number;
  width: number;
  fromPixels?: Array<string>;
};

type GenerateAvatarReturnType = {
  base64: string;
  pixels: Array<string>;
};

declare module 'github-like-avatar-generator' {
  import generateAvatar = require('github-like-avatar-generator');

  function generateAvatar(
    params: GenerateAvatarParams,
  ): GenerateAvatarReturnType;

  export = generateAvatar;
}
