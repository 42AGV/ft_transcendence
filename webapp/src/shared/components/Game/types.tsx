type GameElement = {
  x: number;
  y: number;
  color: string;
};

export type GamePaddle = GameElement & {
  slide: number;
  width: number;
  height: number;
};

export type GameBall = GameElement & {
  vx: number;
  vy: number;
  radius: number;
};

export type Coord = {
  x: number;
  y: number;
};
