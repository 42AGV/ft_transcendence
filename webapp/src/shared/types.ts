export enum Color {
  WARNING = '--color-warning',
  SUBMIT = '--color-submit',
  ONLINE = '--color-online',
  OFFLINE = '--color-offline',
  LIGHT = '--color-light',
  BACKGROUND = '--color-background',
  DARK = '--color-dark',
}
