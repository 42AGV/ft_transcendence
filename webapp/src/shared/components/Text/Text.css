.text-warning {
  color: var(--color-warning);
}

.text-submit {
  color: var(--color-submit);
}

.text-online {
  color: var(--color-online);
}

.text-offline {
  color: var(--color-offline);
}

.text-light {
  color: var(--color-light);
}

.text-background {
  color: var(--color-background);
}

.text-dark {
  color: var(--color-dark);
}

.text-game {
  background: var(--gradient-game);
  background-clip: text;
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
}
