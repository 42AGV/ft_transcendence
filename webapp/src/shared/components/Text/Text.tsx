import './Text.css';

export enum TextType {
  HEADER = 'h1',
  TITLE = 'h3',
  SUBTITLE = 'h5',
  DESCRIPTION = 'p',
}

export enum TextColor {
  WARNING = 'text-warning',
  SUBMIT = 'text-submit',
  ONLINE = 'text-online',
  OFFLINE = 'text-offline',
  LIGHT = 'text-light',
  BACKGROUND = 'text-background',
  DARK = 'text-dark',
  GAME = 'text-game',
}

export enum TextSize {
  EXTRA_SMALL = '0.75rem',
  SMALL = '0.875rem',
  SMALL_MEDIUM = '1rem',
  MEDIUM = '1.25rem',
  MEDIUM_LARGE = '1.5rem',
  LARGE = '2.5rem',
  EXTRA_LARGE = '3rem',
}

export enum TextWeight {
  REGULAR = '400',
  MEDIUM = '500',
  BOLD = '700',
}

type TextProps = {
  children: string;
  type: TextType;
  color?: TextColor;
  size?: TextSize;
  weight?: TextWeight;
};

export default function Text({
  type,
  size = TextSize.MEDIUM,
  color = TextColor.DARK,
  weight = TextWeight.REGULAR,
  children,
}: TextProps) {
  const TextTag = type as React.ElementType;

  return (
    <TextTag style={{ fontSize: size, fontWeight: weight }} className={color}>
      {children}
    </TextTag>
  );
}
