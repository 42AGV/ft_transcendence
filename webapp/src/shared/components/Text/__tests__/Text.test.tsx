import { render, screen } from '@testing-library/react';
import { create } from 'react-test-renderer';
import { Color } from '../../../types';
import Text, { TextSize, TextWeight } from '../Text';

test('renders default text', () => {
  render(<Text>Hello World!</Text>);

  const textElement = screen.getByText(/hello world!/i);
  expect(textElement).toBeInTheDocument();
  expect(textElement).toHaveStyle({
    fontSize: TextSize.MEDIUM,
    fontWeight: TextWeight.REGULAR,
  });
});

test('renders custom text', () => {
  render(
    <Text size={TextSize.EXTRA_LARGE} weight={TextWeight.BOLD}>
      Hello World!
    </Text>,
  );

  const textElement = screen.getByText(/hello world!/i);
  expect(textElement).toBeInTheDocument();
  expect(textElement).toHaveStyle({
    fontSize: TextSize.EXTRA_LARGE,
    fontWeight: TextWeight.BOLD,
  });
});

test('renders correctly with default text', () => {
  const tree = create(<Text>Hello World!</Text>).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly with custom text', () => {
  const tree = create(
    <Text
      size={TextSize.EXTRA_LARGE}
      color={Color.LIGHT}
      weight={TextWeight.BOLD}
    >
      Hello World!
    </Text>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
