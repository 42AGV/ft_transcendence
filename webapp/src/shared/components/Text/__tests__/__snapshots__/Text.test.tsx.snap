// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with custom text 1`] = `
<h3
  className="text-light"
  style={
    Object {
      "fontSize": "3rem",
      "fontWeight": "700",
    }
  }
>
  Hello World!
</h3>
`;

exports[`renders correctly with default text 1`] = `
<h3
  className="text-dark"
  style={
    Object {
      "fontSize": "1.25rem",
      "fontWeight": "400",
    }
  }
>
  Hello World!
</h3>
`;
